content/blog/usethis-2-0-0/index.Rmd

```diff
@@ -0,0 +1,258 @@
+
+This diagram shows the different ways that usethis might interact with GitHub:
+
+1. As a Git server, via gert, using either the HTTPS or SSH protocol
+1. As a web service, via gh, using HTTPS (and OAuth)
+
+```{r echo  = FALSE, fig.align='center', out.width="80%"}
+knitr::include_graphics("pat-kills-both-birds.png")
+```
+
+The cheerful orange ovals indicate why we recommend HTTPS as your Git protocol:
+Once you set up your GitHub personal access token (PAT), usethis, gert, and gh (and possibly other packages) will all be able to find and use this common credential.
+(If you are an SSH person, you need to set up a GitHub PAT for work that involves the GitHub API, in addition to the SSH keys needed for Git work.)
+
+You may have noticed that command line Git remembers your HTTPS credentials, after you've provided them once[^credential-caching].
+Git has an internal interface for storing and retrieving HTTPS credentials from system-specific helpers, such as the macOS Keychain and Windows Credential Manager.
+This interface is exposed for use by other applications in the [`git credential`](https://git-scm.com/docs/git-credential) utility
+Both gert and gh (and, therefore, usethis) now use this utility to retrieve a PAT suitable for a specific `host`.
+
+[^credential-caching]: In our experience, on both macOS and Windows, recent Git versions come with credential caching that works out-of-the-box. If this is not your experience, it's a good reason to update Git.
+
+In the previous section, we explained how the `host` is now automatically discovered from Git remotes.
+And we've just explained that we now look up the PAT based on the `host`.
+Together, this means usethis no longer needs any explicit PAT management and finishes explaining why so many credential-, token-, and host-related functions and arguments have been deprecated.
+This is also a major reason why GitHub Enterprise "just works" now.
+
+The new connection to the system-specific Git credential store also means we no longer need to set `GITHUB_PAT` in our `.Renviron` startup files.
+It is a better security practice anyway to avoid storing such secrets in plain text, whenever possible.
+
+```
+BLAH=foo
+GITHUB_PAT=xyz  # <-- WE SUGGEST YOU REMOVE THIS LINE FROM .Renviron
+WUT=yo
+OTHER_WEB_SERVICE=super-secret-very-powerful-token
+```
+
+The `host`-specific PAT is now retrieved from the Git credential store upon first need.
+Note that the PAT **is** still cached in an environment variable for reuse during the remainder of the current R session.
+
+Our full recommendations for getting and configuring a PAT are in the new vignette [Managing Git(Hub) Credentials](https://usethis.r-lib.org/articles/articles/git-credentials.html).
+
+### Git protocol
+
+When usethis configures a new Git remote, it must choose a protocol, either HTTPS or SSH.
+The new default in this situation is HTTPS, because we think HTTPS is a better choice, for most users and projects (rationale is above).
```

Comment:
```suggestion
The new default in this situation is HTTPS, because as we explained above, we think HTTPS is the best choice for most users and projects.
```